src/sentry/event_manager.py

```diff
@@ -1636,39 +1636,44 @@ def _save_aggregate_new(
 
     group_processing_kwargs = _get_group_processing_kwargs(job)
 
-    # Background grouping is a way for us to get performance metrics for a new
-    # config without having it actually affect on how events are grouped. It runs
-    # either before or after the main grouping logic, depending on the option value.
-    maybe_run_background_grouping(project, job)
-
+    primary_grouping_config, primary_hashes = run_primary_grouping(project, job, metric_tags)
     secondary_grouping_config, secondary_hashes = maybe_run_secondary_grouping(
         project, job, metric_tags
     )
 
-    primary_grouping_config, primary_hashes = run_primary_grouping(project, job, metric_tags)
+    all_hashes = extract_hashes(primary_hashes) + extract_hashes(secondary_hashes)
+
+    grouphashes = [
+        GroupHash.objects.get_or_create(project=project, hash=hash)[0] for hash in all_hashes
+    ]
+
+    existing_grouphash = find_existing_grouphash_new(grouphashes)
+
+    if existing_grouphash:
```

Comment:
I don't know the underlying types, but if we want the condition exactly the same we might want to use `is not None`, so `[]`, `{}`, `""` or whatever don't skip to `create_group_with_grouphashes`

Reply:
A valid point, but in this case it'll be a `GroupHash` object or `None` - no way to get anything else that's falsy, I don't think.